packages/core/src/app/shipping/Shipping.test.tsx

```diff
@@ -0,0 +1,163 @@
+            {} as CheckoutSelectors,
+        );
+
+        jest.spyOn(checkoutService, 'loadShippingOptions').mockResolvedValue(
+            {} as CheckoutSelectors,
+        );
+
+        jest.spyOn(checkoutService, 'deleteConsignment').mockResolvedValue({} as CheckoutSelectors);
+
+        jest.spyOn(checkoutState.data, 'getCart').mockReturnValue({
+            ...getCart(),
+            lineItems: {
+                physicalItems: [
+                    {
+                        ...getPhysicalItem(),
+                        quantity: 3,
+                    },
+                ],
+            },
+        } as Cart);
+
+        jest.spyOn(checkoutState.data, 'getShippingAddress').mockReturnValue(getShippingAddress());
+
+        jest.spyOn(checkoutState.data, 'getBillingAddress').mockReturnValue(undefined);
+
+        jest.spyOn(checkoutState.data, 'getConfig').mockReturnValue(getStoreConfig());
+
+        jest.spyOn(checkoutState.data, 'getShippingAddressFields').mockReturnValue(
+            getAddressFormFields(),
+        );
+
+        jest.spyOn(checkoutState.data, 'getCustomer').mockReturnValue({
+            ...getCustomer(),
+            addresses: [],
+        });
+
+        jest.spyOn(checkoutState.data, 'getConsignments').mockReturnValue([getConsignment()]);
+
+        jest.spyOn(checkoutState.data, 'getCheckout').mockReturnValue(getCheckout());
+
+        jest.spyOn(checkoutService, 'updateBillingAddress').mockResolvedValue(
+            {} as CheckoutSelectors,
+        );
+        jest.spyOn(checkoutService, 'updateCheckout').mockResolvedValue({} as CheckoutSelectors);
+        jest.spyOn(checkoutService, 'updateShippingAddress').mockResolvedValue(
+            {} as CheckoutSelectors,
+        );
+
+        ComponentTest = (props) => (
+            <CheckoutProvider checkoutService={checkoutService}>
+                <LocaleContext.Provider value={localeContext}>
+                    <ExtensionProvider checkoutService={checkoutService}>
+                        <Shipping {...props} />
+                    </ExtensionProvider>
+                </LocaleContext.Provider>
+            </CheckoutProvider>
+        );
+    });
+
+
+    describe('when new multishipping ui is enabled', () => {
+        beforeEach(async () => {
+            jest.spyOn(checkoutState.data, 'getConfig').mockReturnValue({
+                ...getStoreConfig(),
+                checkoutSettings: {
+                    ...getStoreConfig().checkoutSettings,
+                    hasMultiShippingEnabled: true,
+                    features: {
+                        ...getStoreConfig().checkoutSettings.features,
+                        "PROJECT-4159.improve_multi_address_shipping_ui": true,
+                    },
+                },
+            });
+        });
+
+        it.only('opens confirmation dialog on clicking the link and calls onToggleMultiShipping when confirm is clicked', async () => {
```

Comment:
Is the `only` intentional?

Reply:
No, not intentional. Thanks for catching this 🙇‍♀️